### ISSUE TITLE
Support for QT5

### ISSUE THREAD

**Issue by vrguru** [torch/qtlua#19]
I see that qtlua only support QT4.8 or lower? Is there a plan for supporting later QT versions?
For example, QT4.8 only comes in precompiled versions for vs2010 which is...old.

**Comment by leonbottou** [comment]
To be exact, it should be ported to both Qt5 and recent versions of lua.
I just have a hard time finding the time to do it.

It would be go to estimate how many people use qtlua these days...

- L.

**Comment by soumith** [comment]
Indeed, it's a humongous effort with little benefit in terms of adoption.

**Comment by darmie** [comment]
I need qtLua badly, but most of other things my project would depend on need qt5 or higher.

**Comment by wrq** [comment]
If this could support QT5, then it would be possible to use Lua to make SailfishOS smartphone apps. 

I mean, there's obviously more to it than that, but that's one of the big obstacles.

**Comment by floringogianu** [comment]
QT4 will most likely not be supported anymore on macOS Sierra, see Homebrew/homebrew-core#5216 . Only QT5 from this point onwards.

**Comment by zhangxiangxiao** [comment]
I use this a lot. I do not like web-based UI stuff. But I am fine with staying with Qt 4.8 as well.

**Comment by farleylai** [comment]
Qt is definitely a way to cross-platform application development and Qt5 evolves into a nearly full-stack application framework. With Torch, it would be made easier to develop full-fledged AI enabled applications. Perhaps most of the users are research based and the industry either integrates with their own proprietary extension or simply waits for the new updates.
